### PR TITLE
Release note fragment improvements

### PR DESCRIPTION
**Pull request by scop**
## Proposed change

This adds ability to preserve a fragment identifier possibly in the URL when accessing `/latest-release-notes/#something`, and adds identifiers for 0.105.x releases.

We can take advantage of this in the HA release note links, will file a related UI PR.

Somewhat torn whether this is current or next material, but can obviously rebase on request.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by klaasnicolaas** [comment]
There are a lot of unrelated commits, Your are trying to merge changes from a `next` branch into a `current` branch. Please rebase your branch / fork to the `current` branch.

File: source/_posts/2020-02-05-release-105.markdown [hunk 1]
```diff
@@ -483,7 +483,7 @@ Hats over your heart for these shuttered integrations. Pour one out for:
 
 - __Hook__ *Removed* - The `hook` integration has been removed due to the shutdown of the connected service. All servers that have been providing this service will be turned off on 31 Jan 2020. - ([@springstan] - [#31046])
 
-## Release 0.105.1 - February 5
+## <a id="release-0.105.1"></a>Release 0.105.1 - February 5
```
Comment (balloob) [inline source/_posts/2020-02-05-release-105.markdown:486]: All release notes already come with an anchor defined by markdown. We've been moving away from trying to do custom markdown. 

It's also generated here fyi https://github.com/home-assistant/hass-release/blob/master/hassrelease/changelog.py#L152

Comment (scop) [reply source/_posts/2020-02-05-release-105.markdown:486]: The problem with the already existing automatic markdown anchors is that the title for which is generated includes the date, e.g. the "February 5" part above, and this ends up as part of the anchor (`release-01051---february-5` for the above). That date is not available in the UI where the release notes link is generated, all we have there is the version.

**Comment by frenck** [comment]
I rather that we change the titles we generate from the release generator (e.g., move the date in the content of the section).

**Comment by scop** [comment]
No objections to moving the date to the content here :)

Failing that, I guess the release note pages could have a javascript snippet that takes the hash, e.g. `#release-0.105.1` and tries to transform+locate the corresponding markdown anchor for that and scroll there.

**Pull request closed without merging**